.github/workflows/build-db.yml

```
@@ -43,27 +43,27 @@ jobs:
         if: ${{ matrix.platform == 'darwin-arm64' }}
         run: |
           cd package/db
-          FODB_MACHINE=arm64 FODB_SYSTEM=Darwin python setup.py bdist_wheel
+          FODB_MACHINE=arm64 FODB_SYSTEM=Darwin python -Im build
       - name: Build Darwin x86_64 wheel
         if: ${{ matrix.platform == 'darwin-x86_64' }}
         run: |
           cd package/db
-          FODB_MACHINE=x86_64 FODB_SYSTEM=Darwin python setup.py bdist_wheel
+          FODB_MACHINE=x86_64 FODB_SYSTEM=Darwin python -Im build
       - name: Build Windows 32 wheel
         if: ${{ matrix.platform == 'windows-32' }}
         run: |
           cd package/db
-          FODB_MACHINE=32 FODB_SYSTEM=Windows python setup.py bdist_wheel
+          FODB_MACHINE=32 FODB_SYSTEM=Windows python -Im build
       - name: Build Windows x86_64 wheel
         if: ${{ matrix.platform == 'windows-x86_64' }}
         run: |
           cd package/db
-          FODB_MACHINE=x86_64 FODB_SYSTEM=Windows python setup.py bdist_wheel
+          FODB_MACHINE=x86_64 FODB_SYSTEM=Windows python -Im build
       - name: Build sdist
         if: ${{ matrix.platform == 'sdist' }}
         run: |
           cd package/db
-          python setup.py sdist
+          python -Im build
```

Small detail, we only want `sdist` built here so linux machines are able to download the correct MongoDB.
```suggestion
          python -Im build --sdist # linux machines must build from an sdist
```

---

Yes, the smoke test is valuable. The only thing that can't happen is an upload of a Linux wheel. In the latest `fiftyone-db` [release](https://pypi.org/project/fiftyone-db/#files) you can see that Mac and Windows wheels are published, but not Linux. This is required so various Linux distributions (e.g. Ubuntu, RHEL, etc.) are supported.

So passing `--sdist` avoids building the Linux wheel. It is also ok to omit `--sdist` to smoke test that the Linux wheel can be built. But we just don't want to publish it to PyPI in the final [workflow step](https://github.com/voxel51/fiftyone/blob/233f7c6c87c7caf5f16f5445fd75483b02c37f0e/.github/workflows/build-db.yml#L120)

---

Indeed you are right, it would upload the created wheel as well. Also, the smoke test is actually already done for other distributions, so if the sdist build is not good, the other builds will fail.
Done

---

FWIW you should look into `cibuildwheel` — it is able to cover all platform-specific wheels, including GNU/Linux (it makes standards-compliant manylinux and musllinux variants). That's what the wider community defaults to nowadays. Shipping sdists and forcing the end-users to compile on install is considered a hack.